pydicom/sr/srdict.py

```diff
@@ -0,0 +1,253 @@
+            raise AssertionError(msg)
+        else:
+            scheme = matches[0]
+            concept = concepts[scheme][name]
+            # Almost always only one code per concepts name
+            if len(concept) == 1:
+                code, val = list(concept.items())[0]
+            else:
+                matches = [(code, val) for code, val in concept.items()
+                           if self.cid in val[1]]
+                if len(matches) > 1:
+                    # Should never happen, but check in case
+                    msg = "{} had multiple code matches for cid{}".format(name, cid)
+                    raise AssertionError(msg)
+                code, val = matches[0]
+            return Code(
+                value=code,
+                meaning=val[0],
+                scheme_designator=scheme
+            )
+
+    @property
+    def concepts(self):
+        if self._concepts:
+            return self._concepts
+        self._concepts = {name: getattr(self, name) for name in self.dir()}
+        return self._concepts
+
+    def __repr__(self):
+        heading = "CID{}\n".format(self.cid)
+        concepts = [self.repr_format.format(name, concept)
+                    for name, concept in self.concepts.items()]
+        return heading + "\n".join(concepts)
+
+    def __str__(self):
+        heading = "CID{}\n".format(self.cid)
+        fmt = self.str_format
+        line2 = self.str_format.format("Business name", "value",
+                                       "scheme", "meaning")
+        lines = "".join(fmt.format(name, *concept)
+                        for name, concept in self.concepts.items())
+        return heading + line2 + lines
+
+    def dir(self, *filters):
+        """Return an alphabetical list of SR identifiers based on a partial
+        match.
+
+        Intended mainly for use in interactive Python sessions.
+
+        Parameters
+        ----------
+        filters : str
+            Zero or more string arguments to the function. Used for
+            case-insensitive match to any part of the SR keyword.
+
+        Returns
+        -------
+        list of str
+            The matching SR keywords. If no filters are
+            used then all keywords are returned.
+        """
+        allnames = set(chain(*cid_concepts[self.cid].values()))
+        return _filtered(allnames, filters)
+
+    def __contains__(self, code):
+        """Checks whether a given code is a member of the context group.
+
+        Parameters
+        ----------
+        code: Union[pydicom.sr.coding.Code, pydicom.sr.coding.CodedConcept]
+            coded concept
+
+        Returns
+        -------
+        bool
+            whether CID contains `code`
+
+        """
+        coded_concept = CodedConcept(
+            value=code.value,
+            scheme_designator=code.scheme_designator,
+            meaning=code.meaning
+        )
+        return any([coded_concept == concept for concept in self.concepts.values()])
+
```

Comment:
List brackets should be removed so is a generator instead of creating entire list.